### ISSUE TITLE
Local variables does not supported

### ISSUE THREAD

**Issue by bagulin** [dart-lang/angular_analyzer_plugin#161]
In template I got two errors: Undefinde name `item`
```
<div *ngFor="let item of items" [ngClass]="getItemClass(item)">
    {{item.title}}
</div>
```

**Comment by MichaelRFairhurst** [comment]
Thanks for trying it out!

The plugin should support local variables, and I think I know why its not in this example.

Are you using a pub transformer? It will appear as `platform_directives` in your `pubspec.yaml`. Unfortunately our plugin doesn't yet support pub transformers, so you have to add `CORE_DIRECTIVES` to the `directives: const [...]` list. Only once that is done will our plugin pick up on the `ngFor` and set the local variable correctly.

Let me know if that's the reason this is failing for you, and if our stuff is still useful with that workaround in place! We are unfortunately still in prerelease so these things can still happen.

**Comment by bagulin** [comment]
Thank you. You are right, it works with explicit using `NgFor` in directives list but not with `COMMON_DIRECTIVES`. 

Do you have any roadmap for release? We are planning to use this plugin.

**Comment by matanlurey** [comment]
@MichaelRFairhurst:

It would be nice to get `COMMON/CORE_DIRECTIVES` working, but I think supporting `platform_directives` is very low priority. It adds a project-specific configuration to your tool which is something (IMO) we shouldn't worry about until we are feature complete.

**Comment by MichaelRFairhurst** [comment]
Great! We're glad it looks useful, and that that fixed it for you.

The main thing barring an official release is waiting for dynamic
pluggability from the analysis server. I don't know if we have a formal
projected timeframe for that, though I am hoping that is done by the end of
the quarter. At the moment, as you saw, it requires replacing a binary the
in SDK which makes things like upgrading it later a pain. And certainly
installation is not as easy either.

Until then our main focus is internal use so that makes pub transformers
lower on the schedule. Same with a windows line ending bug we've seen.

But aside from that, we sure hope the plugin is useful elsewhere for those
brave enough to make it through the install. It seems decently stable. I
would be maybe able to increase priority some on pub transformers if you
and others are in that camp.

It also may need an API to turn on and off, if you want to help us with
ideas we can comment on the technical viability. Ideally we could analyze
the pubspec, but maybe it could go in `.analysis_options`. And of course
PRs are welcome!

**Comment by MichaelRFairhurst** [comment]
Duplicate of #81, once again, thanks for filing! These notes will be referenced there and help us prioritiize
